src/math/sqrt_int16.c

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+/* Include Files */
+#include <sof/math/sqrt.h>
+
+/* Defines Constant*/
+#define SQRT_WRAP_SCHAR_BITS 0xFF
+
+/* Function Definitions */
+/*
+ *  Square root
+ *
+ *  Y = SQRTLOOKUP_INT16(U) computes the square root of
+ *  U using lookup tables.
+ *  Range of u is [0 to 65535]
+ *  Range of y is [0 to 4]
+ * +------------------+-----------------+--------+--------+
+ * | u		      | y (returntype)	|   u	 |  y	  |
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * | 16 | 12  |	 0    | 16  | 12 |   1	| 4.12	 | 2.14	  |
```

Comment:
It's not resolved, there's still from y Qformat 2.14. Maybe you intended it to be 2.14 but the code in my tests returns 4.12.